src/components/NewMessageModal.vue

```diff
@@ -36,158 +33,117 @@ export default {
+				return t('mail', 'Outbox draft')
+			}
+			return t('mail', 'New message')
 		},
-		templateMessageId() {
-			return this.$store.getters.messageComposerOptions?.templateMessageId
+		composerMessage() {
+			return this.$store.getters.composerMessage
 		},
 		composerData() {
-			logger.debug('composing a new message or handling a mailto link', {
-				threadId: this.$route.params.threadId,
-			})
-
-			let accountId
-			// Only preselect an account when we're not in a unified mailbox
-			if (this.$route.params.accountId !== 0 && this.$route.params.accountId !== '0') {
-				accountId = parseInt(this.$route.params.accountId, 10)
-			}
-			if (this.templateMessageId !== undefined) {
-				if (this.original.attachments.length) {
-					showWarning(t('mail', 'Attachments were not copied. Please add them manually.'))
-				}
-
-				return {
-					accountId: this.original.accountId,
-					to: this.original.to,
-					cc: this.original.cc,
-					subject: this.original.subject,
-					body: this.originalBody,
-					originalBody: this.originalBody,
-				}
-			}
-
-			return {
-				accountId,
-				to: this.stringToRecipients(this.$route.query.to),
-				cc: this.stringToRecipients(this.$route.query.cc),
-				subject: this.$route.query.subject || '',
-				body: this.$route.query.body ? detect(this.$route.query.body) : html(''),
-			}
+			return this.$store.getters.composerMessage?.data
+		},
+		forwardedMessages() {
+			return this.composerMessage?.options?.forwardedMessages ?? []
 		},
-	},
-	created() {
-		this.fetchOriginalMessage()
 	},
 	methods: {
-		stringToRecipients(str) {
-			if (str === undefined) {
-				return []
-			}
-
-			return [
-				{
-					label: str,
-					email: str,
-				},
-			]
-		},
 		async saveDraft(data) {
-			if (data.draftId === undefined && this.draft) {
-				logger.debug('draft data does not have a draftId, adding one', {
-					draft: this.draft,
-					data,
-					id: this.draft.databaseId,
-				})
-				data.draftId = this.draft.databaseId
-			}
-			const dataForServer = {
-				...data,
-				to: data.to.map(this.recipientToRfc822).join(', '),
-				cc: data.cc.map(this.recipientToRfc822).join(', '),
-				bcc: data.bcc.map(this.recipientToRfc822).join(', '),
-				body: data.isHtml ? data.body.value : toPlain(data.body).value,
-			}
-			const { id } = await saveDraft(data.account, dataForServer)
+			if (this.composerMessage.type === 'outbox') {
+				if (data.draftId === undefined && this.draft) {
+					logger.debug('draft data does not have a draftId, adding one', {
+						draft: this.draft,
+						data,
+						id: this.draft.databaseId,
+					})
+					data.draftId = this.draft.databaseId
+				}
+				const dataForServer = {
+					...data,
+					body: data.isHtml ? data.body.value : toPlain(data.body).value,
+				}
+				await this.$store.dispatch('outbox/updateMessage', { message: dataForServer, id: this.composerData.id })
+			} else {
+				if (data.draftId === undefined && this.draft) {
```

Comment:
@GretaD @st3iny this code didn't make any sense but I just left it as is. There was no `this.draft` in the old component either. This will always be undefined.